job/server/src/main/java/alluxio/job/plan/transform/CompactDefinition.java

```diff
@@ -96,16 +96,17 @@ private boolean shouldIgnore(URIStatus status) {
     Map<WorkerInfo, ArrayList<CompactTask>> assignments = Maps.newHashMap();
     int maxNumFiles = config.getMaxNumFiles();
     long groupMinSize = config.getMinFileSize();
-    if (totalFileSize / groupMinSize > maxNumFiles) {
-      groupMinSize = Math.round(totalFileSize / maxNumFiles);
-    }
 
     if (!files.isEmpty() && config.getPartitionInfo() != null) {
       // adjust the group minimum size for source compression ratio
       groupMinSize *= COMPRESSION_RATIO.get(
           config.getPartitionInfo().getFormat(files.get(0).getName()));
     }
 
+    if (totalFileSize / groupMinSize > maxNumFiles) {
```

Comment:
Wouldn't this override the `groupMinSize` variable and discard the compression ratio? 
what was the observed symptom in testing?

Reply:
The problem with the current behavior is that when you have a CSV input and you say I want 100 files with groupMinSize small enough that I should be able to get 100 files, you instead get 20 files.

Reply:
compressionRatio should not be downsizing maxNumFiles.

Reply:
ah makes sense.